### ISSUE TITLE
@NaturalOrder on a collection of immutable generated types

### ISSUE THREAD

**Issue by Stephan202** [immutables/immutables#525]
It seems I hit a limitation of the sanity check performed for the `@NaturalOrder` and `@ReverseOrder` annotations. Consider the following class:

```java
@Value.Immutable
@Value.Style(typeAbstract = "Abstract*", typeImmutable = "*")
public abstract class AbstractSortedMultisetWrapper {
    @Value.NaturalOrder // Works.
    public abstract ImmutableSortedMultiset<AbstractElem> getAbstractElems();

    @Value.NaturalOrder // Causes error.
    public abstract ImmutableSortedMultiset<Elem> getElems();

    // The problem is/may also be exhibited if the element type is declared
    // in a file of its own.
    @Value.Immutable
    public static abstract class AbstractElem implements Comparable<AbstractElem> {
        public abstract int getValue();

        @Override
        public final int compareTo(AbstractElem other) {
            return getValue() - other.getValue();
        }
    }
}
```

This fails to compile, triggering the following error:
```
error: @Value.Natural requires that a (multi)set's elements or a map's key are Comparable
```

Stepping through the code, it seems happens because the `Elem` type isn't known (`TypeKind.ERROR`) at the moment that `ValueAttribute#introspectType()` is invoked. As a result `ValueAttribute#isComparableKey()` yields `false`.

So my question is: is there any way we can make sure the element type is resolved in time, or can we relax the check such that it doesn't complain about unresolvable types?

**Comment by elucash** [comment]
probably, the check can be disabled if a type is unresolved, there is no way to analyze not-yet-generated types

**Comment by Stephan202** [comment]
Thanks for the super-fast response! I had already started an investigation along those lines; the result is in PR #526.

**Comment by Stephan202** [comment]
I'll close this issue since the PR has been merged. Thanks again :).

**Comment by elucash** [comment]
this was released in 2.4.1. Thank you!
